src/GovUk.Education.ExploreEducationStatistics.Data.Model/LocalAuthority.cs

```diff
@@ -12,6 +12,11 @@ public LocalAuthority(string? code, string? oldCode, string? name) : base(code,
             OldCode = oldCode;
         }
 
+        public string GetCacheKey()
+        {
+            return $"{GetType().Name}:{Code ?? string.Empty}:{OldCode ?? string.Empty}:{Name ?? string.Empty}";
```

Comment:
Why do we not use `GetCodeOrFallback()` here?  Are you explicitly wanting to add to the cache key the fact that the `Code` isn't present?

Reply:
If so, might be worth a comment here to explain why.

Reply:
Yes, and because we need a separate code and old code token inside the cache key, rather than using `GetCodeOrFallback` to differentiate locations with the same code and name but different old code.

Imagine we had `$"{GetType().Name}:{GetCodeOrFallback()}:{Name ?? string.Empty}";` or `$"{GetType().Name}:{GetCodeOrOldCodeIfEmpty()}:{Name ?? string.Empty}";`

Two imported LA rows with the same Code, and Name but different old codes will hit the same cache entry as the old code will be ignored.

Unit test `ImporterLocationServiceTests.FindOrCreate_UniqueByLocalAuthorityOldCode` added by this PR now covers this case.

Prior to this PR this exists as a bug in Prod, see the current code on line 135:

https://github.com/dfe-analytical-services/explore-education-statistics/blob/1090a7b9438aeb6f6f69934097b40c8ef1a05b6c/src/GovUk.Education.ExploreEducationStatistics.Data.Processor/Services/ImporterLocationService.cs#L135

Reply:
Brill thanks.

Reply:
Comment added in EES-3011.